creator-node/src/utils/strToReplicaSet.ts

```diff
@@ -0,0 +1,23 @@
+export type ReplicaSet = {
+  primary?: string
+  secondary1?: string
+  secondary2?: string
+}
+
+/**
+ * Converts a string of endpoints to a ReplicaSet type.
+ * @param {string} replicaSetStr The comma-separated string of endpoints like:
+ *                      "http://cn_primary.co,http://cn_secondary1.co,http://cn_secondary2.co"
+ * @return {ReplicaSet} a ReplicaSet type of the string split by comma, where the first index is the primary and the other 2 indexes are secondaries
+ */
+export function strToReplicaSet(replicaSetStr: string): ReplicaSet {
+  const replicaSetArr = replicaSetStr.split(',')
+  if (replicaSetArr?.length !== 3) {
+    throw new Error(`Expected 3 replicas in array but got: ${replicaSetArr}`)
```

Comment:
enforcement of length = 3 will cause some issues, e.g. the thing that vicky pointed out about some users having replicaset length 2

that might be ok here, but just flagging

Reply:
thanks for flagging - I made [this card](https://linear.app/audius/issue/CON-364/change-getuserreplicasetendpointsfromdiscovery-to-use-sp-ids) after our discussion today to make SP IDs be the source of truth. we shouldn't be using `creator_node_endpoint` in the first place, but we do it in a couple other places already so will have to update them all

Reply:
yep agreed, thx